### PR TITLE
Refined dbfs implementation and tested it in a real cluster

### PR DESCRIPTION
**Pull request by nils-braun**
Thanks for the nice start @martindurant! 
I did some small refactorings and cleanings, fixed some small bugs and tested it on a "real" Azure databricks cluster. I can list, create and read files :-) 

Still to do:
* fix the open TODO
* add tests
* add documentation
* add handling for errors, exceptions or edge-cases

### CODE REVIEW

File: fsspec/implementations/dbfs.py [hunk 1]
```diff
@@ -1,66 +1,115 @@
...
+
+    def get_data(self, path, start, end):
+        r = self.get_from_api(
+            "read",
+            json={"path": path, "offset": start, "length": end - start},
+        )
+        return base64.b64decode(r["data"])
+
+    def post_to_api(self, endpoint, json):
+        r = self.session.post(os.path.join(self.url, "2.0/dbfs", endpoint), json=json)
```
Comment (martindurant) [inline fsspec/implementations/dbfs.py:97]: os.path.join will do the wrong thing on windows. urlparse has a join, but I think "/" or just `+` are fine.

File: fsspec/implementations/dbfs.py [hunk 2]
```diff
@@ -1,66 +1,115 @@
...
         r.raise_for_status()
 
+        return r.json()
 
-class DBFile(AbstractBufferedFile):
+    def get_from_api(self, endpoint, json):
```
Comment (martindurant) [inline fsspec/implementations/dbfs.py]: get and post are similar enough, I would make join them, with a `method=` kwarg

File: fsspec/implementations/dbfs.py [hunk 3]
```diff
@@ -1,66 +1,115 @@
...
-                              json={"source_path": path1, "destination_path": path2})
+        if maxdepth:
+            raise NotImplementedError
+
+        self.post_to_api(
+            "move",
+            json={"source_path": path1, "destination_path": path2},
+        )
+
+    def create_handle(self, path, overwrite=True):
```
Comment (martindurant) [inline fsspec/implementations/dbfs.py]: These nicely reflect the real API, but are not meant to be called directly by users, so might prefix with underscore

File: fsspec/implementations/dbfs.py [hunk 4]
```diff
@@ -86,23 +143,17 @@ def __init__(
...
+        self.buffer.seek(0)
+        # TODO: for data in buffer, in 1MB blocks
+        print("Writing data")
+        self.fs.add_data(handle=self.handle, data=self.buffer.getvalue())
         if final:
-            r = self.fs.session.post(self.fs.url + "2.0/dbfs/close",
-                                     json={"handle": self.path})
-            r.raise_for_status
+            print("Closing handle", self.handle)
+            self.fs.close_handle(handle=self.handle)
```
Comment (martindurant) [inline fsspec/implementations/dbfs.py]: Should return `True` to clear the write buffer

**Review by martindurant — COMMENTED** [review]
Some comments so far.
Thanks for the input!

**Comment by nils-braun** [comment]
Thanks for the comments, @martindurant!
The print statements do also need to go away of course ;-)

**Comment by nils-braun** [comment]
I have implemented your proposals and added some more documentation and error handling (handling all the exceptions the API documentation specifies).
The code now also works with large file uploads and downloads. 

The only thing missing are now the tests, but we might want to do this in a second PR.

**Comment by martindurant** [comment]
We should probably have at least a very minimal test first, and then we can iterate.

**Comment by martindurant** [comment]
I made you a collaborator on this fork, so you should be able to push to my branch, if you want @nils-braun
